application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,152 @@
+package org.togetherjava.tjbot.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.GenericInteractionCreateEvent;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.nio.charset.StandardCharsets;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * used to log activities in the mod audit log channel
+ */
+public enum ModAuditLogWriter {
+    ;
+    private static final Color EMBED_COLOR = Color.decode("#3788AC");
+
+    private static final Logger logger = LoggerFactory.getLogger(ModAuditLogWriter.class);
+
+    private static final Predicate<String> isAuditLogChannelName =
+            Pattern.compile(Config.getInstance().getModAuditLogChannelPattern()).asMatchPredicate();
```

Comment:
It is dangerous to do this `static`, the `Config` might not have been loaded yet... (it is loaded shortly after start of the app, not statically)

Reply:
see 54d927a565c43f465dc8dd0c14260970b82ec3c0